evm_arithmetization/src/cpu/kernel/asm/transactions/router.asm

```diff
@@ -47,15 +47,14 @@ global update_txn_trie:
     // stack: txn_rlp_len, value_ptr, txn_counter, num_nibbles, ret_dest
     DUP2 %increment
     // stack: rlp_start=value_ptr+1, txn_rlp_len, value_ptr, txn_counter, num_nibbles, retdest
-    
 
     // and now copy txn_rlp to the new block
     %stack (rlp_start, txn_rlp_len, value_ptr, txn_counter, num_nibbles) -> (
-        @SEGMENT_RLP_RAW, // src addr. ctx == virt == 0
-        rlp_start, @SEGMENT_TRIE_DATA, // swapped dest addr, ctx == 0
+        @SEGMENT_TRIE_DATA, rlp_start, // dest addr, ctx == 0
+        @INITIAL_TXN_RLP_ADDR, // src addr
         txn_rlp_len, // mcpy len
```

Comment:
`txn_rlp_len` but now I realized is not the case

Reply:
yeah sorry I changed the stack macro because there was a useless SWAP just after, may have been confusing

Reply:
No I'm just confused about writing `trie_data_len` at `SEGMENT_RLP_RAW` since it's also at `GLOBAL_METADATA_TRIE_DATA_SIZE`

Reply:
This is for the transaction trie update. We're copying the txn RLP from `SEGMENT_RLP_RAW` (starting at `INITIAL_TXN_RLP_ADDR`), to `SEGMENT_TRIE_DATA`, with offset `rlp_start`. So that we don't need to do any thing and we can compute directly the new node to insert in the txn trie.

Reply:
But INITIAL_TXN_RLP_ADDR = 2, and rlp_start is obtained by reading the global metadata value, right? So why do we need to write the length in RlpRaw as well?

Reply:
I'm confused. Are we talking about the highlighted stack operation here or something else?

Reply:
Ah sorry, I think my comment (at least) should have been in the other conversation